dev-docs/howtoguides/troubleshoot_apt_news_security_confinement.md

```diff
@@ -0,0 +1,215 @@
+
+
+
+## Troubleshooting Apparmor
+
+The Apparmor profile for the `apt-news` service is loaded via the `AppArmorProfile` directive in the unit file `/lib/systemd/system/apt-news.service`:
+```
+  [Service]
+  ...
+  AppArmorProfile=ubuntu_pro_apt_news
+```
+
+This will apply the specified apparmor profile on service startup. If the profile does not exist, the service startup will fail. The actual profile is located in `/etc/apparmor.d/ubuntu_pro_apt_news`, and is loaded into the kernel at package install/upgrade time, or when the system boots.
+
+To verify if the Apparmor profile is causing the issues you are observing, the first troubleshooting attempt should be to put it in "complain" mode. In that mode, it will allow everything, but log if something would have been blocked had the profile been in "enforce" mode.
+
+To place the profile in complain mode, first install the `apparmor-utils` package, if it's not installed already:
+```
+  sudo apt install apparmor-utils
+```
+
+Then run this command:
+```
+  sudo aa-complain /etc/apparmor.d/ubuntu_pro_apt_news
+```
+
+This will both change the profile file to include the `complain` flag, and reload it into the kernel.
+
+Next, keep an eye on the `dmesg` output with something like this:
+```
+sudo dmesg -wT | grep -E 'apparmor=\".*(profile=\"ubuntu_pro_|name=\"ubuntu_pro_)'
+```
+
+And exercise the service. For example, to be sure it will run, first remove some files:
+```
+sudo rm -rf /var/lib/apt/periodic/update-success-stamp /run/ubuntu-advantage /var/lib/ubuntu-advantage/messages/*
+```
+
+And then start the service:
+```
+sudo systemctl start apt-news.service
+```
+
+If you see any logs with `ALLOWED` in them, then that action is something that would have been blocked by the apparmor profile had it not been in "complain" mode, and is something you should add to the apparmor profile.
+
+To make changes to the apparmor profile, edit the `/etc/apparmor.d/ubuntu_pro_apt_news` file, save, and reload the profile with the following command:
+```
+sudo apparmor_parser -r -W -T /etc/apparmor.d/ubuntu_pro_apt_news
+```
+
+Explaining the full syntax of the apparmor profiles is out of scope for this document. You can find more information in the [apparmor.d manpage](https://manpages.ubuntu.com/manpages/noble/man5/apparmor.d.5.html). The Ubuntu Server Guide also has a good introduction on the topic in the [Security - AppArmor](https://ubuntu.com/server/docs/security-apparmor) page.
+
+ATTENTION: be mindful of the differences in Ubuntu Releases regarding the apparmor profile syntax!
+
+
+## Troubleshooting systemd sandboxing
+
+Troubleshooting systemd sandboxing is not as straightforward as with apparmor, because there are no specific logs telling you that a certain action was blocked. It will just be blocked, and it's up to the application to handle it. There is no "system" log to help with troubleshooting the sandbox rules.
```

Comment:
```suggestion
Troubleshooting systemd sandboxing is not as straightforward as with AppArmor, because there are no specific logs telling you that a certain action was blocked. It will just be blocked, and it's up to the application to handle it. There is no "system" log to help with troubleshooting the sandbox rules.
```